lib/child_process.js

```diff
@@ -118,18 +120,26 @@ exports.execFile = function(file /*, args, options, callback*/) {
     env: null
   };
 
-  // Parse the parameters.
+  // Parse the optional positional parameters.
+  var pos = 1;
+  if (pos < arguments.length && Array.isArray(arguments[pos])) {
+    args = arguments[pos++];
+  } else if (pos < arguments.length && arguments[pos] == null) {
+    pos++;
+  }
```

Comment:
Well, the fall through on this particular one is required for the next check. Overall, the currently behavior in v0.12 (and this port of that code) is really just a half measure. We're not actually doing a complete check. For instance, `execFile('ls',{a:1},'test')` goes through without a throw even tho a string is passed in for the callback. Internally, `execFile` just acts as if the callback wasn't provided at all. Likewise, `execFile('ls',[], 'test', function() {})` does not throw either. Nor does `execFile('ls',[],'test','test')`. The only type that is actually checked is the `args`, so that if you pass in `execFile('ls', 'test')`, a `TypeError` will throw.

Given that this is exactly how it behaves in v0.12 currently (for better or worse), I'd be more inclined to land it as is so we can close the convergence loop first, then make additional improvements separately.